crates/walrus-sdk/src/client.rs

```diff
@@ -254,27 +254,44 @@ impl<T: ReadClient> Client<T> {
         SliverData<U>: TryFrom<Sliver>,
     {
         tracing::debug!("starting to read blob");
+
         self.check_blob_id(blob_id)?;
+
         let committees = self.get_committees().await?;
 
-        let certified_epoch = if committees.is_change_in_progress() {
-            tracing::info!("epoch change in progress, reading from initial certified epoch");
-            let blob_status = match blob_status {
-                Some(status) => status,
-                None => {
-                    self.get_blob_status_with_retries(blob_id, &self.sui_client)
-                        .await?
-                }
+        let get_status_fn = |status: Option<BlobStatus>| async move {
+            let status = if let Some(status) = status {
+                status
+            } else {
+                self.get_blob_status_with_retries(blob_id, &self.sui_client)
+                    .await?
             };
-            blob_status
-                .initial_certified_epoch()
-                .ok_or_else(|| ClientError::from(ClientErrorKind::BlobIdDoesNotExist))?
+
+            if BlobStatus::Nonexistent == status {
+                Err(ClientError::from(ClientErrorKind::BlobIdDoesNotExist))
+            } else {
+                Ok(status)
+            }
+        };
+
+        let (certified_epoch, blob_status) = if committees.is_change_in_progress() {
```

Comment:
Nit: Can we rename the `certified_epoch`? In the `unwrap_or_else` of the `else` branch, this is set to some "dummy" value, so this isn't actually always the `certified_epoch`.